src/ApiClient/index.tsx

```diff
@@ -4,48 +4,40 @@
 import {NativeModules} from 'react-native';
 import isURL from 'validator/es/lib/isURL';
 
-import type {
-  RequestOptions,
-  Response,
-  GenericClientInterface,
-  ApiClientInterface,
-  ApiClientConfiguration,
-} from './types';
-
 const {NetworkClient} = NativeModules;
 
 /**
  * Generic client for making GET requests
  */
 class GenericClient implements GenericClientInterface {
-  get = (url: string, options?: RequestOptions) => NetworkClient.get(url, null, options);
+    get = (url: string, options?: RequestOptions): Promise<Response> => NetworkClient.get(url, null, options);
 }
 
 /**
  * Configurable client for consuming a REST API
  */
 class ApiClient implements ApiClientInterface {
-  baseUrl: string;
+    baseUrl: string;
 
-  constructor(baseUrl: string) {
-      this.baseUrl = baseUrl;
-  }
+    constructor(baseUrl: string) {
+        this.baseUrl = baseUrl;
+    }
 
-  getHeaders = (): Promise<object> => NetworkClient.getApiClientHeadersFor(this.baseUrl);
-  addHeaders = (headers: object): void => NetworkClient.addApiClientHeadersFor(this.baseUrl, headers);
+    getHeaders = (): Promise<Headers> => NetworkClient.getApiClientHeadersFor(this.baseUrl);
+    addHeaders = (headers: Headers): Promise<void> => NetworkClient.addApiClientHeadersFor(this.baseUrl, headers);
```

Comment:
not blocking, should we consider returning something here? could it be that we are trying to add a header and that could fail? if the answer is no, then nvm

Reply:
I think eventually yes; it'd be good to return the header objects to prevent subsequent calls - but for now the promise either resolves or rejects (throws errors) so we do get confirmation and that should be okay.